joplin/base/models.py

```diff
@@ -33,7 +34,10 @@ class ServicePage(Page):
     extra_content = StreamField(
         [
             ('content', blocks.RichTextBlock(features=WYSIWYG_FEATURES, help_text='Write any additional content describing the service')),
-            ('application_block', custom_blocks.SnippetChooserBlockWithAPIGoodness('base.ApplicationBlock')),
+            ('application_block', custom_blocks.SnippetChooserBlockWithAPIGoodness('base.ApplicationBlock', icon='site')),
+            ('map_block', custom_blocks.SnippetChooserBlockWithAPIGoodness('base.Map', icon='site')),
```

Comment:
what is the `SnippetChooserBlockWithAPIGoodness`?

Reply:
By default, wagtail only serializes the top-level amount of data for snippets within `StreamBlock`s. I made this thing to serialize all the fields of the snippet:

https://github.com/cityofaustin/joplin/blob/1ad7ae0f4ee9ec50e6223056da31cbb840c0168c/joplin/base/blocks.py#L4

That was for the wagail REST API. I used that idea to build out how we serialize snippets in graphql:

https://github.com/cityofaustin/joplin/blob/1ad7ae0f4ee9ec50e6223056da31cbb840c0168c/joplin/api/schema.py#L15-L23